playbooks/roles/mattermost/templates/config.json.j2

```diff
@@ -302,9 +384,70 @@
         "Password": "",
         "EnableIndexing": false,
         "EnableSearching": false,
+        {% if {{ MATTERMOST_VERSION }} != '4.0.0' %}
+            "PostIndexReplicas": 1,
+            "PostIndexShards": 1,
+            "AggregatePostsAfterDays": 365,
+            "PostsAggregatorJobStartTime": "03:00",
+            "IndexPrefix": "",
+            "LiveIndexingBatchSize": 1,
+            "BulkIndexingTimeWindowSeconds": 3600,
+            "RequestTimeoutSeconds": 30,
         "Sniff": true
     },
     "DataRetentionSettings": {
-        "Enable": false
+        {% if {{ MATTERMOST_VERSION }} == '4.0.0' %}
+            "Enable": false
+        {% else %}
+            "EnableMessageDeletion": false,
+            "EnableFileDeletion": false,
+            "MessageRetentionDays": 365,
+            "FileRetentionDays": 365,
+            "DeletionJobStartTime": "02:00"
```

Comment:
Jinja2 templates aren't white space sensitive like Python. `{% if %}` blocks must be ended with `{% endif %}`.

You should have gotten a parsing error when you tested this, but if not it definitely isn't going to work the way you expect.

Reply:
I went over it more carefully and corrected it. I also ran the playbook to ensure the config file gets parsed correctly.